libr/reg/reg.c

```diff
@@ -202,6 +202,40 @@ static int regcmp(RRegItem *a, RRegItem *b) {
 	return offa > offb;
 }
 
+R_API void r_reg_rerole(RReg *reg) {
+	int i, j, index;
+	RListIter *iter;
+	RRegItem *r;
+	RList *all = r_list_newf (NULL);
+	for (i = 0; i < R_REG_TYPE_LAST; i++) {
+		r_list_foreach (reg->regset[i].regs, iter, r) {
+			r->role = R_REG_NAME_LAST;
+		}
+	}
+	for (i = 0; i < R_REG_TYPE_LAST; i++) {
+		r_list_foreach (reg->regset[i].regs, iter, r) {
+			if (r->role != R_REG_NAME_LAST) {
```

Comment:
Maybe decouple these nested loops? Make a separate small functions probably?

Reply:
im just caring about correctness of the result